docs/TuxMutableClass.md

```diff
@@ -0,0 +1,188 @@
+
+```javascript
+    var mutableClassProps = {};
+    mutableClassProps.mutableTraits = {
+      props: 'text',
+      state: 'editing'
+    };
+
+    // OR
+    mutableClassProps.mutableTraits = {
+      props: ['text', 'timestamp'],
+      state: 'editing'
+    };
+
+    // OR it is possible to specify the hierarchy of keys to search through to improve performance of the one time deep search/provide greater specificity regarding the required property
+    mutableClassProps.mutableTraits = {
+      props: [
+        ['message', 'text'],
+        ['message', 'timestamp']
+      ],
+      state: 'editing'
+    };
+```
+
+#### <a id="mutableClassProps-render"></a>Parameter - `mutableClassProps.render` - type: FUNCTION - required [#](#mutableClassProps-render)
+Since this is an implementation of `React.createClass` a `render` method is required.
+
+#### <a id="createMutableClass-mutableClass"></a>Return - 'mutableClass' - type: OBJECT [#](#createMutableClass-mutableClass)
+Returns a React class augmented with special Tux mixins.
+
+## <a id="TuxMutableClass-Guidelines"></a>TuxMutableClass Guidelines [#](#TuxMutableClass-Guidelines)
+We recommend using the mutable class in situations where you are rendering many components of the same class, and only some of the state and props in those components will ever update.
+
+A specific example is if you have tens or hundreds of message components rendering on the screen. The mutable class would be beneficial at that point because most of those message components can update, but will not necessarily update with great frequency. Therefore having components that will check to see if they should re-render in the virtual DOM can result in a performance boost with the more messages that are on the screen. If someone edits the text of a message, instead of re-rendering all of the other messages in the virtual DOM with the mutable class only the one message that has updated will do so.
+
+## <a id="TuxMutableClass-Complete-Example"></a>TuxMutableClass Complete Example [#](#TuxMutableClass-Complete-Example)
+Taking all that we have learned so far let's take a look at a more realistic example.
+
+```javascript
+    var React = require('tux/React');
+
+    var Message = React.createMutableClass({
+      mutableTraits: {
+        props: [
+          ['message', 'text']
+        ],
+        state: 'editing'
+      },
+
+      getInitialState: function () {
+        return {
+          editing: false
+        };
+      },
+
+      deleteMessage: function(e) {
+        e.preventDefault();
+        this.nearestOwnerProps.deleteMessage(this.props.message.id); // the mutableClass is an extension of the owneeClass. In this example the Message component is a child of a component of an ownerClass component which has a deleteMessage function.
+      },
+
+      updateMessage: function (message) {
+        this.nearestOwnerProps.updateMessage(message, this.props.message.id);
```

Comment:
This wouldn't work. You should instead pass a `ref` attribute to the `input` component you are rendering and then use it  here with:

```
    var message = this.YOUR INPUT REF.getDOMNode().value
```

Also. the argument passed into `updateMessage` would be the event, so you should use that to call `e.preventDefault()`.